src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -448,11 +442,10 @@ protected virtual CompletionItem GetBetterItem(CompletionItem item, CompletionIt
         private class DisplayNameToItemsMap : IEnumerable<CompletionItem>, IDisposable
         {
             // We might need to handle large amount of items with import completion enabled,
-            // so use a dedicated pool to minimize array allocations.
-            // Set the size of pool to a small number 5 because we don't expect more than a
-            // couple of callers at the same time.
-            private static readonly ObjectPool<Dictionary<string, object>> s_uniqueSourcesPool
-                = new(factory: () => new(), size: 5);
+            // so use a dedicated pool to minimize array allocations. Set the size of pool to a small
+            // number 5 because we don't expect more than a couple of callers at the same time.
+            private static readonly ObjectPool<Dictionary<string, object>> s_uniqueSourcesPool = new(factory: () => new(), size: 3);
+            private static readonly ObjectPool<List<CompletionItem>> s_sortListPool = new(factory: () => new(), size: 3);
```

Comment:
💡 explicit type would be better than `new()` here for readability

Reply:
fixed